src/machine.rs

```diff
@@ -86,24 +85,49 @@ impl<'tcx> Frame<'tcx> {
         }
     }
 
-    fn set_local(&mut self, local: Local, val: Vec<u8>) {
-        self.locals[local] = Some(val)
+    fn set_local(&mut self, local: Local, bytes: Vec<u8>) {
+        self.locals[local] = bytes
     }
 
     fn get_local(&self, local: Local) -> &[u8] {
-        match self.locals[local] {
-            Some(ref v) => v.as_slice(),
-            None => panic!("Local does not exist")
-        }
+        self.locals[local].as_slice()
+    }
+
+    fn get_local_field(&self, local: Local, ptr: Ptr) -> &[u8] {
+        let local = self.get_local(local);
+        &local[ptr.addr..ptr.size]
+    }
+
+    fn set_local_field(&mut self, local: Local, ptr: Ptr, bytes: Vec<u8>) {
+        assert_eq!(ptr.size, bytes.len());
+        let end = ptr.addr + ptr.size;
+        self.locals.get_mut(local)
+                   .unwrap()
+                   .splice(ptr.addr..end, bytes.into_iter());
     }
 }
 
 /// Represents a pointer into some kind of memory.
 // TODO: Extend to work with static constructs.
 #[derive(Debug)]
 pub enum Address {
-    Heap(usize), // pointer to offset in memory.
-    Local(Local, usize),
+    Heap(Ptr), // pointer to offset in memory.
+    Local(Local, Option<Ptr>), // Optional ptr for specific field access
+}
+
+#[derive(Debug)]
+pub struct Ptr {
+    pub addr: usize,
+    pub size: usize,
+}
+
+impl Ptr {
```

Comment:
Add a comment on this type.

Reply:
I'm not against commenting up structs. I just thought this one was too obvious ;)

Reply:
Fair enough.